### ISSUE TITLE
Update tFPDF

### ISSUE THREAD

**Issue by aheckler** [woocommerce/sensei-certificates#117]
We are on version 1.24. The new version seems to be here:

https://github.com/DocnetUK/tfpdf

Latest is 2.0.6.

**Comment by annezazu** [comment]
563666-zd User running into errors seemingly due to this not being up to date:

```
Deprecated: Methods with the same name as their class will not be constructors in a future version of PHP; tFPDF has a deprecated constructor in /wp-content/plugins/sensei-certificates/lib/tfpdf/tfpdf.php on line 13

Notice: Object of class WP_Error could not be converted to int in /wp-includes/post.php on line 4965

Deprecated: Methods with the same name as their class will not be constructors in a future version of PHP; TTFontFile has a deprecated constructor in /wp-content/plugins/sensei-certificates/lib/tfpdf/font/unifont/ttfonts.php on line 38 
FPDF error: Unsupported image type: /wp-content/uploads/
```

@pgk  Is this related? Any chance we can update this?

**Comment by andrewjohnstrong** [comment]
I really need tfpdf updated, about to deliver sensei project using certificates to a client and now I can't :( I have read that this is related to php 7 which I am using.

**Comment by PDowney** [comment]
Food for thought, it currently works using HHVM. You could always just move to that for the time being.




---- On Tue, 18 Apr 2017 18:11:19 -0700 Andrew Strong &lt;notifications@github.com&gt; wrote ----




I really need tfpdf updated, about to deliver sensei project using certificates to a client and now I can't :( I have read that this is related to php 7 which I am using.

**Comment by andrewjohnstrong** [comment]
@PDowney what is HHVM? and how would I move to it?

**Comment by PDowney** [comment]
HHVM is an alternative to PHP 7. It runs just as fast too. I don't know anything about the stack you're using, but if it's a self managed VPS switching to HHVM would be fairly simple.

**Comment by andrewjohnstrong** [comment]
Unfortunately that's not an option for me, but thank you for the idea.

**Comment by pgk** [comment]
@annezazu this has been addresed in https://github.com/woocommerce/sensei-certificates/commit/80e01664b36153f3bc5e3e11e39cd4625bae016f. Upgrading tfpdf is a good idea for a future release though

**Comment by pgk** [comment]
kinda related to #120 (to be released)

**Comment by jenhooks** [comment]
@alexsanford -- this is kind of old, but it came up again in 1695209-HC. Wondering if it is a quick win to fix it?

**Comment by alexsanford** [comment]
Deprioritizing this. See https://github.com/woocommerce/sensei-certificates/pull/149#issuecomment-406325530

If we decide to tackle this in the future (assuming the backwards compatibility issues are resolved), #149 may be a good reference.

**Comment by donnapep** [comment]
Looks like we're going to need to figure out a solution here. See #159.
